### ISSUE TITLE
training_loss and val_loss become nan after longtime training

### ISSUE THREAD

**Issue by echo385** [ultralytics/ultralytics#21064]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Why does the training end directly become 0, including the loss also becomes nan?

<img width="1448" alt="Image" src="https://github.com/user-attachments/assets/c42f02a9-1a49-4c15-872f-d7a5a590711b" />
<img width="1151" alt="Image" src="https://github.com/user-attachments/assets/fda3d1e4-22aa-4f79-906d-0ff662291d55" />
![Image](https://github.com/user-attachments/assets/3a81b1b4-b03c-40b9-96bc-f632b8d1bc79)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @echo385, thank you for reaching out and sharing your training results with Ultralytics 🚀! Your question is important to us.

We suggest checking out the [Docs](https://docs.ultralytics.com/) for helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, as well as answers to many common training issues.

Since this may be a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE)? This will help our team investigate the `nan` loss values you’re seeing. Please include:
- The exact training command used
- Model and dataset details
- Relevant logs and, if possible, a snippet of your data or code

If you're customizing your training or dataset, make sure to share sample images and logs, and reference our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to rule out common pitfalls.

Join the Ultralytics community for more support:
- For real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussion: [Discourse](https://community.ultralytics.com/)
- For sharing and learning: [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Before proceeding, please ensure you're using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response 🦾. An Ultralytics engineer will review your issue and provide further assistance soon!

**Comment by Y-T-G** [comment]
What's the command you used for training?

**Comment by echo385** [comment]
> What's the command you used for training?

`yolo detect train data=datasets/silk_data_V6.yaml model=yolo11l.yaml device=1 imgsz=1280 epochs=3000 batch=16 lr0=0.0001 lrf=0.01 optimizer='Adam' resume=True cache=True workers=16`

**Comment by Y-T-G** [comment]
How many images do you have?

**Comment by echo385** [comment]
> How many images do you have?

16985 images for train, 4247 images for val

**Comment by glenn-jocher** [comment]
The NaN loss issue is likely caused by training instability from your extremely high epoch count (3000) and potentially overfitting. Try reducing epochs to 100-300 and add early stopping with `patience=50` to prevent overfitting. Also consider increasing your batch size to 32-64 if GPU memory allows, as the small batch size with large image size (1280) can cause gradient instability.

**Comment by echo385** [comment]
> The NaN loss issue is likely caused by training instability from your extremely high epoch count (3000) and potentially overfitting. Try reducing epochs to 100-300 and add early stopping with `patience=50` to prevent overfitting. Also consider increasing your batch size to 32-64 if GPU memory allows, as the small batch size with large image size (1280) can cause gradient instability.

thanks!  when I use 4090D with 8gpus to train my data, the same training params, but the 4090D crashed rapidly,  why is that?

**Comment by glenn-jocher** [comment]
Multi-GPU training crashes are often due to insufficient memory per GPU or synchronization issues. With 8 GPUs, try using `device=[0,1,2,3,4,5,6,7]` and reduce your batch size to `batch=8` (total effective batch will be 8x8=64), or use `batch=-1` for automatic batch size selection. Also scale your learning rate proportionally - try `lr0=0.0008` (8x your current rate) for the increased effective batch size.
